tests/plot/test_PlotSpikes.py

```diff
@@ -0,0 +1,252 @@
+    <Include file="Simulation.xml"/>
+
+    <Include file="izhikevich2007_network.nml"/>
+
+    <Simulation id="example_izhikevich2007network_sim" length="1000ms" step="0.1ms" target="IzNet" seed="123">  <!-- Note seed: ensures same random numbers used every run -->
+        <EventOutputFile id="Spikes_file__IzhPop0" fileName="{spike_file_name}" format="ID_TIME">
+            <EventSelection id="0" select="IzhPop0[0]" eventPort="spike">
+            </EventSelection>
+            <EventSelection id="1" select="IzhPop0[1]" eventPort="spike">
+            </EventSelection>
+            <EventSelection id="2" select="IzhPop0[2]" eventPort="spike">
+            </EventSelection>
+            <EventSelection id="3" select="IzhPop0[3]" eventPort="spike">
+            </EventSelection>
+            <EventSelection id="4" select="IzhPop0[4]" eventPort="spike">
+            </EventSelection>
+            <EventSelection id="5" select="IzhPop0[5]" eventPort="spike">
+            </EventSelection>
+        </EventOutputFile>
+        <EventOutputFile id="Spikes_file__IzhPop1" fileName="{spike_file_name}" format="ID_TIME">
+            <EventSelection id="6" select="IzhPop1[0]" eventPort="spike">
+            </EventSelection>
+            <EventSelection id="7" select="IzhPop1[1]" eventPort="spike">
+            </EventSelection>
+            <EventSelection id="8" select="IzhPop1[2]" eventPort="spike">
+            </EventSelection>
+            <EventSelection id="9" select="IzhPop1[3]" eventPort="spike">
+            </EventSelection>
+            <EventSelection id="10" select="IzhPop1[4]" eventPort="spike">
+            </EventSelection>
+        </EventOutputFile>
+
+
+
+    </Simulation>
+
+</Lems>
+    """
+
+        lems_file = tempfile.NamedTemporaryFile(mode="w", delete=False, dir=".")
+        print(lems_contents.format(spike_file_name=spike_file_name), file=lems_file)
+        lems_file.flush()
+        lems_file.close()
+
+        for i in range(6):
+            file_name = f"IzhPop0[{i}].dat"
+            with open(file_name, "w") as f:
+                for j in range(100):
+                    print(f"{j*10} {i}", file=f)
+
+        for i in range(5):
+            file_name = f"IzhPop1[{i}].dat"
+            with open(file_name, "w") as f:
+                for j in range(100):
+                    print(f"{j*10} {i+6}", file=f)
+
+        pyplts.plot_spikes_from_lems_file(
+            lems_file_name=lems_file.name,
+            base_dir=".",
+            show_plots_already=True,
+            save_spike_plot_to="spikes-test-from-lems-file.png",
+            rates=False,
+            rate_window=50,
+            rate_bins=500,
+        )
+        self.assertIsFile("spikes-test-from-lems-file.png")
+
+        pyplts.plot_spikes_from_lems_file(
+            lems_file_name=lems_file.name,
+            base_dir=".",
+            show_plots_already=False,
+            save_spike_plot_to="spikes-test-from-lems-file-2.png",
+            rates=True,
```

Comment:
we'll have to look at the implementation of `rates` later, not sure it's doing the right thing. Don't worry about it for this PR, though.